context/app/routes_browse.py

```diff
@@ -48,39 +49,26 @@ def details(type, uuid):
     entity = client.get_entity(uuid)
     actual_type = entity['entity_type'].lower()
 
-    # Redirect to primary dataset if this is
-    # - a support entity (e.g. an image pyramid)
-    # - a processed or component dataset
-    is_support = actual_type == 'support'
-    is_processed = entity.get('processing') != 'raw' and actual_type == 'dataset'
-    is_component = entity.get('is_component', False) is True
-    if (is_support or is_processed or is_component):
-        supported_entity = client.get_entities(
-            'datasets',
-            query_override={
-                "bool": {
-                    "must": {
-                        "terms": {
-                            "descendant_ids": [uuid]
-                        }
-                    }
-                }
-            },
-            non_metadata_fields=['hubmap_id', 'uuid']
-        )
+    if (should_redirect_entity(entity)):
+        earliest_dataset = find_raw_dataset_ancestor(client, entity.get('ancestor_ids'))
 
         pipeline_anchor = entity.get('pipeline', entity.get('hubmap_id')).replace(' ', '')
         anchor = quote(f'section-{pipeline_anchor}-{entity.get("status")}').lower()
 
-        if len(supported_entity) > 0:
-            return redirect(
-                url_for('routes_browse.details',
-                        type='dataset',
-                        uuid=supported_entity[0]['uuid'],
-                        _anchor=anchor,
-                        redirected=True,
-                        redirectedFromId=entity.get('hubmap_id'),
-                        redirectedFromPipeline=entity.get('pipeline')))
+        # Check whether the oldest found ancestor exists and is of an expected type. 404 is
+        # preferable to a page that shows only a support entity or processed/component dataset.
+        if not earliest_dataset or should_redirect_entity(earliest_dataset[0]):
```

Comment:
Why do we need to call `should_redirect_entity` again?

Reply:
This was originally to check that the retrieved entity was not actually a processed/support dataset - with the updated query, this shouldn't be necessary.